### ISSUE TITLE
Getting "custom attribute panicked" message when using `[#Object]` helper

### ISSUE THREAD

**Issue by biot023** [async-graphql/async-graphql#409]
## Expected Behavior
It should turn my query and mutation roots into `async_graphql::ObjectType`s.

## Actual Behavior
I get this error message:

> error: custom attribute panicked                         
>   --> src/schema.rs:12:1                                 
>    |                                                     
> 12 | #[Object]                                           
>    | ^^^^^^^^^                                           
>    |                                                     
>    = help: message: assertion failed: index <= self.len()
>                                                         
> error: custom attribute panicked                         
>   --> src/schema.rs:22:1                                 
>    |                                                     
> 22 | #[Object]                                           
>    | ^^^^^^^^^                                           
>    |                                                     
>    = help: message: assertion failed: index <= self.len()

## Steps to Reproduce the Problem

My code:

```rust
use async_graphql::*;                                                         
                                                                              
use crate::Context;                                                           
                                                                              
pub struct QueryRoot;                                                         
                                                                              
#[Object]                                                                     
impl QueryRoot {                                                              
                                                                              
    async fn ping() -> &str {                                                 
        "Pong"                                                                
    }                                                                         
}                                                                             
                                                                              
pub struct MutationRoot;                                                      
                                                                              
#[Object]                                                                     
impl MutationRoot {                                                           
                                                                              
    async fn index_item() -> async_graphql::Result<bool> {                    
        unimplemented!()                                                      
    }                                                                         
}                                                                             
                                                                              
pub fn create_schema() -> Schema<QueryRoot, MutationRoot, EmptySubscription> {
    Schema::new(QueryRoot, MutationRoot, EmptySubscription)                   
}                                                                             
```

## Specifications

- Version: 2.5.2
- Platform: Ubuntu Linux
- Subsystem: 18.04

**Comment by sunli829** [comment]
The `&self` parameter is required.
This is a bug and no useful error message is given.

```rust

#[Object]                                                                     
impl QueryRoot {                                                              
                                                                              
    async fn ping(&self) -> &str {                                                 
        "Pong"                                                                
    }                                                                         
}     
```

**Comment by sunli829** [comment]
Released in `v2.5.3`
